server/sonar-server/src/test/java/org/sonar/server/issue/ws/SearchActionComponentsTest.java

```diff
@@ -19,242 +19,597 @@
+  }
+
+  @Test
+  public void search_by_project_uuid() throws Exception {
+    ComponentDto project = insertComponent(ComponentTesting.newPublicProjectDto(otherOrganization1, "P1").setDbKey("PK1"));
+    ComponentDto file = insertComponent(newFileDto(project, null, "F1").setDbKey("FK1"));
+    IssueDto issue = IssueTesting.newDto(newRule(), file, project).setKee("82fd47d4-b650-4037-80bc-7b112bd4eac2");
+    db.issueDao().insert(session, issue);
+    session.commit();
+    indexIssues();
+    indexPermissions();
+
+    wsTester.newRequest()
+      .setParam(IssuesWsParameters.PARAM_PROJECT_UUIDS, project.uuid())
+      .execute()
+      .assertJson(this.getClass(), "search_by_project_uuid.json");
+
+    wsTester.newRequest()
+      .setParam(IssuesWsParameters.PARAM_PROJECT_UUIDS, "unknown")
+      .execute()
+      .assertJson(this.getClass(), "no_issue.json");
+
+    wsTester.newRequest()
+      .setParam(IssuesWsParameters.PARAM_COMPONENT_UUIDS, project.uuid())
+      .execute()
+      .assertJson(this.getClass(), "search_by_project_uuid.json");
+
+    wsTester.newRequest()
+      .setParam(IssuesWsParameters.PARAM_COMPONENT_UUIDS, "unknown")
+      .execute()
+      .assertJson(this.getClass(), "no_issue.json");
+  }
+
+  @Test
+  public void search_since_leak_period_on_project() throws Exception {
+    ComponentDto project = insertComponent(ComponentTesting.newPublicProjectDto(otherOrganization2, "P1").setDbKey("PK1"));
+    ComponentDto file = insertComponent(newFileDto(project, null, "F1").setDbKey("FK1"));
+    db.snapshotDao().insert(session,
+      newAnalysis(project)
+        .setPeriodDate(parseDateTime("2015-09-03T00:00:00+0100").getTime()));
+    RuleDto rule = newRule();
+    IssueDto issueAfterLeak = IssueTesting.newDto(rule, file, project)
+      .setKee(UUID_EXAMPLE_01)
+      .setIssueCreationDate(parseDateTime("2015-09-04T00:00:00+0100"))
+      .setIssueUpdateDate(parseDateTime("2015-10-04T00:00:00+0100"));
+    IssueDto issueBeforeLeak = IssueTesting.newDto(rule, file, project)
+      .setKee(UUID_EXAMPLE_02)
+      .setIssueCreationDate(parseDateTime("2014-09-04T00:00:00+0100"))
+      .setIssueUpdateDate(parseDateTime("2015-10-04T00:00:00+0100"));
+    db.issueDao().insert(session, issueAfterLeak, issueBeforeLeak);
+    session.commit();
+    indexIssues();
+    indexPermissions();
+
+    wsTester.newRequest()
+      .setParam(IssuesWsParameters.PARAM_COMPONENT_UUIDS, project.uuid())
+      .setParam(IssuesWsParameters.PARAM_SINCE_LEAK_PERIOD, "true")
+      .execute()
+      .assertJson(this.getClass(), "search_since_leak_period.json");
+  }
+
+  @Test
+  public void search_since_leak_period_on_file_in_module_project() throws Exception {
+    ComponentDto project = insertComponent(ComponentTesting.newPublicProjectDto(defaultOrganization, "P1").setDbKey("PK1"));
+    ComponentDto module = insertComponent(newModuleDto(project));
+    ComponentDto file = insertComponent(newFileDto(module, null, "F1").setDbKey("FK1"));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Fix or remove this skipped unit test [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1607)